### DISCUSSION TITLE
Can someone please explain difference in throughput

### DISCUSSION THREAD

**Discussion by janput** [NginxProxyManager/nginx-proxy-manager#3940] [Q&A]
Hi,
I'm running speedtest behind nginx proxy manager in a docker container using bridge networking. Both containers run on the same docker daemon and all test are performed on the same machine running the docker daemon.
I see a big difference in speed when using the local docker port (both http and https) vs NPM (https enabled).
Can someone please explain why and if there are any settings to tweak the throughput.

When passing via NPM:
![NPM](https://github.com/user-attachments/assets/ecb70f06-2348-414a-8f95-87550338373d)

When using the speedtest docker http port:
![docker-http](https://github.com/user-attachments/assets/bb81d532-155f-4a6e-879e-41fd106f074d)

When using the speedtest docker https port:
![docker-https](https://github.com/user-attachments/assets/e9215cd4-d756-49b1-acfa-a770e06fd2d7)

I understand that the https connection introduces a little overhead, but I don't understand why NPM is so much slower.
Any help would be appreciated

**Comment by ecxsystems** [comment]
Hi Team!

I have a similar issue where I have a website with some 19MB PDF attachments.  They can take 2 mins to download.  Without NPM, it is nearly instant.

Is there some tuning document around performance?
